### ISSUE TITLE
Error when parsing async arrow functions with destructed default parameters.

### ISSUE THREAD

**Issue by Alxandr** [babel/babel#852]
The following works as you would expect:

```javascript
var per2 = ({id, ext, name, model = false, view = false}) => {
  
};
```

However, if you add async in front of it, you get a syntax error:

```javascript
var per2 = async ({id, ext, name, model = false, view = false}) => {
  
};
```

```
repl: Unexpected token (1:20)
> 1  | var a = async ({a, b=2}) => {};
     |                     ^
```

**Comment by sebmck** [comment]
Yep, this will be fixed pending babel/acorn-babel#27. Async arrow functions are currently parsed extremely weirdly due to the ambiguity.

**Comment by sebmck** [comment]
Rewrote the async function parsing. Released as of 4.4.4.
